docs/contributing.md

```diff
@@ -27,3 +27,15 @@ Then, please [create a new issue in the GitHub repository](https://github.com/mo
 Be sure to include as much detail as possible including step-by-step descriptions, specific examples, screenshots or mockups, and reasoning for why the enhancement might be worthwhile.
 
 Please keep in mind, by opening an issue we provide no guarantee the enhancement will be implemented.
+
+## How to Contribute Code
+
+Before you get started writing code, be sure what you plan to work on is something we'll be able to accept. The easiest way is to look through out list of [good first issues][good-first-issues] and find something that sounds interesting.
+
+If there's something else you'd like to work on, just add a comment in the relevant issue and we'll be happy to discuss your plans. If you have an idea that doesn't have an issue at all, be sure to [file an issue](#how-to-request-enhancements) first.
+
+Once you have a patch ready:
+- Make sure all tests are passing and the code you have written is tested. You can run tests in XCode by selecting the `uispecs` target and hitting CMD-U.
```

Comment:
One way this could be read is "don't submit a PR until you've written a test"; is this what we want to gate on?  I think gating merging a PR on tests is appropriate, but I'm not sure about gating the submission of a PR on tests.

If a lowered entry bar is acceptable, what about the following as a starting point?

> - Make sure all tests are passing and try to add tests for the code you've written. You can run tests in XCode by selecting the `uispecs` target and hitting `CMD-U`.

Reply:
Ok, I'm fine with that.